File: iOSApp/Collegefeed/Collegefeed/PostCreateViewController.m
//
//  PostCreateViewController.m
//  TheCampusFeed
//
//  Created by Patrick Sheehan on 1/8/15.
//  Copyright (c) 2015 Appuccino. All rights reserved.
//

#import "PostCreateViewController.h"
#import "College.h"
#import "DataController.h"
#import "Post.h"
#import "Shared.h"
#import "ToastController.h"

@interface PostCreateViewController ()

@property (strong, nonatomic) College *college;

@end

@implementation PostCreateViewController

#pragma mark - View life cycle

- (id)initWithDataController:(DataController *)controller
{
    self = [super initWithNibName:@"CreateViewController" bundle:nil];
    if (self)
    {
        self.dataController = controller;
        [self setModalPresentationStyle:UIModalPresentationCustom];
        [self setTransitioningDelegate:self];
        return self;
    }
    return nil;
}
- (void)loadView
{
    [super loadView];
}
- (void)viewDidLoad
{
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}
- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    if (self.college != nil && self.college.name != nil)
    {
        self.subtitleLabel.text = [NSString stringWithFormat:@"Posting to %@", self.college.name];
    }
    
    [self.titleLabel setText:@"New Post"];
    self.cameraButtonWidth.constant = 40;
    
    [self.view setNeedsLayout];
    [self.view layoutIfNeeded];
}
- (void)viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (BOOL)assign:(College *)newCollege
{
    if (newCollege != nil)
    {
        self.college = newCollege;
        [self.view setNeedsLayout];
        return YES;
    }
    
    return NO;
}

- (IBAction)submit:(id)sender
{
    NSString *message = self.messageTextView.text;
    if ([Post withMessageIsValid:message])
    {
        NSLog(@"Post with message = %@ is valid", message);
        
        if (self.college != nil || self.college.collegeID > 0)
        {
            [self.dataController submitPostToNetworkWithMessage:message
                                                  withCollegeId:self.college.collegeID
                                                      withImage:self.imageView.image];
        }
        else
        {
            NSLog(@"College ID needed but not found for Post submission");
            [Shared queueToastWithSelector:@selector(toastPostFailed)];
        }
        
        [self dismiss:self];
    }
    else
    {
        NSLog(@"Post with message = %@ is NOT valid", message);
        [Shared queueToastWithSelector:@selector(toastInvalidPost)];
    }
}

@end

File: iOSApp/Collegefeed/Collegefeed/PostCreateViewController.swift
//
//  PostCreateViewController.swift
//  TheCampusFeed
//
//  Created by Patrick Sheehan on 1/8/15.
//  Copyright (c) 2015 Appuccino. All rights reserved.
//

import UIKit

class PostCreateViewController: CreateViewController {

    var college: College?

    override func viewDidLoad() {
        super.viewDidLoad()
        self.titleLabel.text = "New Post"
        self.cameraButtonWidth.constant = 40;
        
        self.view.setNeedsLayout()
        self.view.layoutIfNeeded()
    }
    
    func assign(newCollege: College) {
        if let c = newCollege as College? {
            self.subtitleLabel.text = "Posting to \(c.name)"
            self.college = c
        }
    }
    
    override func submit(sender: AnyObject!) {
        
        if let message = self.messageTextView.text {
            if Post.withMessageIsValid(message) {
                println("Post message is valid")
                if let collegeId = self.college!.collegeID as Int?{
                    self.dataController.submitPostToNetworkWithMessage(message, withCollegeId:collegeId, withImage:self.imageView!.image)
                }
                else
                {
                    println("College ID needed but not found for Post submission")
                    Shared.queueToastWithSelector(Selector("toastPostTooShort"))
                }
            }
            else
            {
                println("Post message is invalid")
            }
        }
    }
}
